syntax = "proto3";

package auth;

import "google/protobuf/empty.proto";

service RegistrationService{
  rpc Register(RegisterRequest) returns (RegisterResponse);
}

message RegisterResponse{
  string access_token = 1;
  string refresh_token = 2;
  int64 expires_in = 3;
}

message RegisterRequest{
  string email = 1;
  string name = 2;
  string password = 3;
}

service UserService{

  rpc GetUserProfile(GetUserRequest) returns (GetUserResponse);
  rpc UpdatePassword(UpdatePasswordRequest) returns (google.protobuf.Empty);
}

message UpdatePasswordRequest{
  string email = 1;
  string name = 2;
  string current_password = 3;
  string new_password = 4;
}

message GetUserResponse{
  string email = 1;
  string name = 2;
  string password = 3;
}

message GetUserRequest{
  string email = 1;
  string password = 3;
}


service AuthService {
  rpc Login(LoginRequest) returns (LoginResponse);
  rpc Logout(LogoutRequest) returns (google.protobuf.Empty);
}

message LogoutRequest{
  string access_token = 1;
}

message LoginRequest{
  string email = 1;
  string password = 3;
}

message LoginResponse{
  string access_token = 1;
  string refresh_token = 2;
  int64 expires_in = 3;
}